### ISSUE TITLE
The error message is not helpful at all

### ISSUE THREAD

**Issue by CGQAQ** [ziglang/zig#16725]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Output

https://godbolt.org/z/nqGW753Yq

### Expected Output

Give me the location of my own code

### Actual Output
![image](https://github.com/ziglang/zig/assets/15936231/6d8f52fb-59a4-4961-b260-e97038a2f359)

**Comment by matu3ba** [comment]
```zig
const std = @import("std");

pub fn main() !void {
    const logical_ant = 1;
    const result = 15;

    std.log.err("Ant#{}: Scan result: {02X}\n", .{ logical_ant, result });
    // format string is not correct here ^
}
```

**Comment by dbandstra** [comment]
I ran into this too when upgrading an old codebase. There were some format strings I was using that are no longer valid, and I wasn't getting any trace to tell me where the problem was, only the `@compileError` line in `fmt.zig`.

Here's a reduction (zig 0.11.0) which hopefully demonstrates the same issue (could be oversimplified though):

```zig
pub fn format(comptime x: bool) void {
    _ = x;
    @compileError("error");
}

pub fn main() void {
    format(true);
}
```

This doesn't print any trace, but if you remove `comptime` from the argument, it does.

I found some existing issues on the topic of `@compileError` leaving partial or no trace, but they all seemed to be describing something a little different. E.g. https://github.com/ziglang/zig/issues/14492#issuecomment-1465249012, https://github.com/ziglang/zig/issues/12969, https://github.com/ziglang/zig/issues/14335
